### PR TITLE
Limit genesis building volatile database to 100GB

### LINKED ISSUE #2840: Volatile mdbx opening fails on RISC-V
It is not clear if this is a RISC-V limitation or a limitation specific to the server, but on Scaleway's RV1 cloud servers, compilation fails because of a `Other(12)` MDBX error when generating the genesis blocks. Code 12 stands for ENOMEM, which means "not enough memory".

Through trial and error @jsdanielh found that the virtual mapped database size of 1TB, with which a volatile DB is opened for the build, is the problem. The server only seems to support 171GB, although on another server that was still too much.

For databases during runtime, the client respects a database `size` client config option. But such a config cannot be used for the compilation step. 

I think the volatile database for genesis generation can be reasonably limited to 100GB, which could be passed as `max_size` config option into the database opening method here: https://github.com/nimiq/core-rs-albatross/blob/fab6cbcf6c1b5d80bc35e0b61321b896bea971e2/genesis-builder/src/main.rs#L20

### PR DESCRIPTION
**Pull request by sisou**
Replaces #2842. Please read the discussion there, too!

This is to support platforms that don't support 1TB mapped databases, such as RISC-V in testing. At runtime, the database size can be limited via a config option, but for compilation, no such configuration is possible.

100GB should be more than enough for genesis generation.

## Pull request checklist

- [x] All tests pass. The project builds and runs.
- [x] I have resolved any merge conflicts.
- [x] I have resolved all `clippy` and `rustfmt` warnings.

### CODE REVIEW

File: genesis/build.rs [hunk 1]
```diff
@@ -34,8 +34,12 @@ fn generate_albatross(name: &str, out_dir: &Path, src_dir: &Path) {
     let genesis_config = src_dir.join(format!("{name}.toml"));
     log::info!("genesis source file: {}", genesis_config.display());
 
-    let db =
-        MdbxDatabase::new_volatile(Default::default()).expect("Could not open a volatile database");
+    let db = MdbxDatabase::new_volatile(DatabaseConfig {
+        // Limit the database to 100GB to support platforms with a lower allowed maximum
+        size: Some(0..(100 * GIGABYTE as isize)),
```
Comment (hrxi) [inline genesis/build.rs]:
```suggestion
        size: Some(0..(100 * 1024 * 1024 * 1024)),
```

**Review by hrxi — COMMENTED** [review]
Looks good to me from the functionality side.

Could you maybe add "fixes #2840" to the commit message as well?

File: genesis/build.rs (hunk 1 above)
Comment (sisou) [reply genesis/build.rs]: Can you explain why you think this suggestion is better?

File: database/src/mdbx/database.rs [hunk 2]
```diff
@@ -10,8 +10,8 @@ use crate::{
     Error,
 };
 
-const GIGABYTE: usize = 1024 * 1024 * 1024;
-const TERABYTE: usize = GIGABYTE * 1024;
+pub const GIGABYTE: usize = 1024 * 1024 * 1024;
+pub const TERABYTE: usize = GIGABYTE * 1024;
```
Comment (hrxi) [inline database/src/mdbx/database.rs]: Hmm, sorry, it seems like not all of my comments made it through. I wanted to comment here that I don't think it's good for the database crate to export `GIGABYTE`/`TERABYTE` constants as that seems pretty unrelated to databases.

```suggestion
const GIGABYTE: usize = 1024 * 1024 * 1024;
const TERABYTE: usize = GIGABYTE * 1024;
```

**Review by hrxi — APPROVED** [review]
Thanks!

**Comment by hrxi** [comment]
Rebased for merging.

**Pull request merged**